controllers/discovery/searchUser.js

```diff
@@ -13,12 +13,6 @@ const SearchUser = async (req, res) => {
   const {q, limit = 50, allow_anon_dm} = req.query;
   const {userId} = req;
 
-  if (q.length < 2)
-    return res.status(200).json({
-      success: true,
-      message: 'Your search characters is too few, please input 3 or more characters for search'
-    });
-
   let where_anon_dm = '';
   if (allow_anon_dm) {
     where_anon_dm = 'AND u.allow_anon_dm = :allow_anon_dm';
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The removal of the minimum character length check for the search query could potentially lead to performance issues. If a user enters a very short or empty string, it might result in a large number of results being returned, which could slow down the system. Consider adding a check to ensure that the search query is not empty.
```diff
+  if (!q || q.trim() === '') {
+    return res.status(400).json({
+      success: false,
+      message: 'Search query cannot be empty'
+    });
+  }
```
This will prevent empty searches and maintain the performance of your application.


<!-- This is an auto-generated comment by OSS CodeRabbit -->